backend/hdfs/fs.go

```diff
@@ -35,11 +83,26 @@ func NewFs(ctx context.Context, name, root string, m configmap.Mapper) (fs.Fs, e
 		return nil, err
 	}
 
-	client, err := hdfs.NewClient(hdfs.ClientOptions{
+	options := hdfs.ClientOptions{
 		Addresses:           []string{opt.Namenode},
-		User:                opt.Username,
 		UseDatanodeHostname: false,
-	})
+	}
+
+	if len(opt.ServicePrincipalName) > 0 {
```

Comment:
`if opt.ServicePrincipalName != "" {` would probably be more idiomatic. It compiles to the same code anyway!